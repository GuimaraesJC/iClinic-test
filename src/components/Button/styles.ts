import styled from 'styled-components';

type ButtonProps = {
  buttonWidth: number;
  font: string
}

type FontStyleOptions = {
  [key: string]: {
    fontSize: number;
    lineHeight: number;
  };
}

const fontStyle: FontStyleOptions = {
  'normal': {
    fontSize: 16,
    lineHeight: 20,
  },
  'large': {
    fontSize: 18,
    lineHeight: 22,
  }
}

export const DefaultButton = styled.button<ButtonProps>`
  width: ${props => props.buttonWidth}px;
  height: 56px;
  background: #2c97d1;
  border: none;
  border-radius: 10px;
  color: #fff;
  font-weight: bold;
  font-size: ${props => fontStyle[props.font].fontSize}px;
  line-height: ${props => fontStyle[props.font].lineHeight}px;
  letter-spacing: 0.35em;
  text-transform: uppercase;
`;
