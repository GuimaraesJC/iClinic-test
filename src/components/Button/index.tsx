import * as S from './styles';

type ButtonProps = {
  label: string;
  size: string;
  font: string;
  disabled: boolean;
  onClickFn?: () => void;
}

type ButtonSizeOptions = {
  [key: string]: number;
}

const buttonSize: ButtonSizeOptions = {
  'small': 189,
  'large': 347
}


function Button({ label, size, font, disabled, onClickFn }: ButtonProps) {
  return (
    <S.DefaultButton
      buttonWidth={buttonSize[size]}
      font={font}
      disabled={disabled}
      onClick={onClickFn}
    >
      {label}
    </S.DefaultButton>
  );
}

export default Button;
