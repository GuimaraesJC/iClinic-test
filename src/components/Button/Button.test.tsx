import { render, screen } from '@testing-library/react';

import Button from '.';

const props = {
  size: 'large',
  font: 'normal',
  label: 'default label',
  master: '',
  disabled: false,
  onClick: () => {}
}

describe('<Button />', () => {
  it('should render a black button with yellow text', () => {
    render(<Button {...props} master="Luke Skywalker" />);

    expect(screen.getByText(props.label)).toHaveStyle({
      backgroundColor: '#2a2a2a',
      color: '#fbfe63'
    });
  });

  it('should render a white button with black text', () => {
    render(<Button {...props} master="Darth Vader" />);

    expect(screen.getByText(props.label)).toHaveStyle({
      backgroundColor: '#fff',
      color: '#2a2a2a'
    });
  });

  it('should render a small button', () => {
    render(<Button {...props} size="small" />);

    expect(screen.getByText(props.label)).toHaveStyle({
      width: '189px'
    });
  });

  it('should render a large button', () => {
    render(<Button {...props} size="large" />);

    expect(screen.getByText(props.label)).toHaveStyle({
      width: '347px'
    });
  });

  it('should not be clickable', () => {
    render(<Button {...props} disabled={true} />);

    expect(screen.getByText(props.label)).not.toBeEnabled();
  });

  it('should match the snapshot', () => {
    const { container } = render(<Button {...props} />);

    expect(container).toMatchSnapshot();
  });
});
