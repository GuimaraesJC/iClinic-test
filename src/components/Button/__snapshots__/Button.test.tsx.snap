// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match the snapshot 1`] = `
.c0 {
  width: 347px;
  height: 56px;
  background: #2c97d1;
  border: none;
  border-radius: 10px;
  color: #fff;
  font-weight: bold;
  font-size: 16px;
  line-height: 20px;
}

<div>
  <button
    class="c0"
    label="default label"
  >
    default label
  </button>
</div>
`;
